code.js

```diff
@@ -1,3 +1,37 @@
 function mergesort(array) {
+    var length = array.length;
+    var first = true;
+    var j;
+    var i;
+    if (length <= 1) {
+        return array;
+    } else {
+        for (var size = 1; size < length; size = 2 * size) { // These loops were written by me but ZachRenz helped me understand how to write them
+            j = size;
+            if (first){
+                j = 1;
+                first = false;
+            } 
+            i = 0;
+            while (j < length) {
+                merge(i, j, size, array);
+                i += 2 * size;
+                j += 2 * size;
+            }
+        }
+    }
     return array;
 }
+
+function merge(i, j, size, array) { // This merge function was also written by me minus the for loop
+    while (i < j) {
+        if (array[i] <= array[j]) {
+            i++;
+        } else {
+            for (var k = j + 1; k <= (j - 1 + size) && array[k] < array[i]; k++) { // Chat GPT helped me with getting this specifc piece of logic to get the rest of the code working
```

Comment:
This loop doesn't seem to do anything.

Reply:
The point of the loop is to get a k value that works with the code, so the loop doesnt need to do anything other than that. I need a k value to switch elements in the array as seen in the next line of the code.

Reply:
It still contributes to run time though so I have to calculate this into the runtime

Reply:
Ok, but then it looks like you're only doing a single merge step here. Wouldn't it be more efficient to perform all necessary merge steps, i.e. you don't just find a single index for each call of the merge function?

Reply:
Seems like this is unnecessarily inefficient...

Reply:
Ok I've redone the merge function to include a middle variable so it works with both all the merge steps. I'm still unsure if the runtime is correct so I would appreciate some guidance there if possible